libvast/src/system/index.cpp

```diff
@@ -482,7 +482,12 @@ index_state::collect_query_actors(query_state& lookup,
 void index_state::add_flush_listener(flush_listener_actor listener) {
   VAST_DEBUG("{} adds a new 'flush' subscriber: {}", *self, listener);
   flush_listeners.emplace_back(std::move(listener));
-  detail::notify_listeners_if_clean(*this, *stage);
+  // We may need to call `notify_listeners_if_clean` if the subscription
+  // happens after the data has already completely passed the index, but
+  // we must not to call it before any data at all has arrived it would
+  // create a false positive.
+  if (active_partition.actor != nullptr)
+    detail::notify_listeners_if_clean(*this, *stage);
```

Comment:
Does this need a separate bug fix changelog entry?

Reply:
I think technically it's not really user-visible, because the whole notifying stream stage issues only are visible during integration tests.

Reply:
Fine by me.